collective/pwexpiry/tests/pwexpiryp4.robot

```diff
@@ -0,0 +1,85 @@
+*** Settings ***
+
+Resource  plone/app/robotframework/keywords.robot
+Variables  plone/app/testing/interfaces.py
+
+*** Variables ***
+
+${PORT} =  55001
+${ZOPE_URL} =  http://localhost:${PORT}
+${PLONE_URL} =  ${ZOPE_URL}/plone
+${BROWSER} =  Firefox
+
+*** Keywords ***
+
+Start Browser and Autologin as
+    [arguments]  ${role}
+
+    Open Test Browser
+    Enable Autologin as  $role
+
+Start Browser and Log In as Site Owner
+    Open Test Browser
+    Log In As Site Owner
+    Click Link  link=Home
+
+Go to Site Setup
+    Go to   ${PLONE_URL}/@@overview-controlpanel
+    Wait until location is  ${PLONE_URL}/@@overview-controlpanel
+
+Go to Security Settings
+    Go to   ${PLONE_URL}/@@security-controlpanel
+    Wait until location is  ${PLONE_URL}/@@security-controlpanel
+
+Go to Users and Groups
+    Go to   ${PLONE_URL}/@@usergroup-userprefs
+    Wait until location is  ${PLONE_URL}/@@usergroup-userprefs
+
+Go to Configuration Registry
+    Go to   ${PLONE_URL}/portal_registry
+    Wait until location is  ${PLONE_URL}/portal_registry
+
+Setup PWexpiry Use Username Login
+    Start Browser and Log In as Site Owner
+    Go to Security Settings
+    Select Checkbox  id=form.enable_user_pwd_choice
+    Click Button  id=form.actions.save
+    Go to Configuration Registry
+    Input Text  id=q  validity_period
+    Click Button  Filter
+    Click Link  collective pwexpiry validity_period
```

Comment:
@frapell seems to me you may need to wait here until the link is visible or ass a sleep directive for the same purpose.